tecken/api/views.py

```diff
@@ -628,3 +631,69 @@ def nones_to_zero(obj):
 
     context = {"stats": numbers}
     return http.JsonResponse(context)
+
+
+# Store a result for 10 minutes
+SYMINFO_RESULT_CACHE_TIMEOUT = 600
+
+# Indicates there's nothing in the cache
+NO_VALUE_IN_CACHE = object()
+
+
+@metrics.timer_decorator("lookup_by_syminfo.timing")
+def cached_lookup_by_syminfo(somefile, someid, refresh_cache=False):
+    """Looks up somefile/someid in fileupload data; caches result
+
+    This value is cached.
+
+    :arg somefile: a string that's either a debug_file or a code_file
+    :arg someid: a string that's either a debug_id or a code_id
+    :arg refresh_cache: force a cache refresh
+
+    :returns: dict with keys from lookup_by_syminfo
+
+    """
+    key = f"lookup_by_syminfo::{somefile}//{someid}"
+    ret = cache.get(key, default=NO_VALUE_IN_CACHE)
+    if ret is NO_VALUE_IN_CACHE or refresh_cache is True:
+        ret = FileUpload.objects.lookup_by_syminfo(some_file=somefile, some_id=someid)
+        cache.set(key, ret, SYMINFO_RESULT_CACHE_TIMEOUT)
+        metrics.incr("lookup_by_syminfo.cached", tags=["result:false"])
+    else:
+        metrics.incr("lookup_by_syminfo.cached", tags=["result:true"])
+
+    return ret
+
+
+@metrics.timer_decorator("codeinfo_test")
+@api_require_http_methods(["GET"])
+@set_cors_headers(origin="*", methods="GET")
+def syminfo(request, some_file, some_id):
+    start_time = time.time()
+    ret = cached_lookup_by_syminfo(somefile=some_file, someid=some_id)
+    end_time = time.time()
+    if ret:
+        sym_file = ret["debug_filename"]
+        if sym_file.endswith(".pdb"):
+            sym_file = sym_file[:-4] + ".sym"
+
+        new_url = request.build_absolute_uri(
+            reverse(
+                "download:download_symbol",
+                args=(ret["debug_filename"], ret["debug_id"], sym_file),
+            )
+        )
+
+        return http.JsonResponse(
+            {
+                "debug_id": ret["debug_id"],
+                "debug_filename": ret["debug_filename"],
+                "code_file": ret["code_file"],
+                "code_id": ret["code_id"],
+                "generator": ret["generator"],
+                "url": new_url,
+                "time": end_time - start_time,
```

Comment:
This is goofy--I should ditch this. If I'm doing timings locally, I can wrap the script in a timer and that's good enough. This is in a timer_decorator so we'll get timings in grafana.